apps/web/lib/drag-and-drop.ts

```diff
@@ -0,0 +1,89 @@
+import { DraggableData, DraggableEvent } from "react-draggable";
+
+export interface DragState {
+  // The id of the element that is being dragged
+  dragSourceId: string | null;
+  // The id of the element that is currently being hovered over
+  dragTargetId: string | null;
+  // The position of the elements being dragged such that on drag over, we can revert the position.
+  initialX: number;
+  initialY: number;
+}
+
+export interface DragAndDropFunctions {
+  handleDragStart: (e: DraggableEvent, data: DraggableData) => void;
+  handleDrag: (e: DraggableEvent) => void;
+  handleDragEnd: () => void;
+  dragState: DragState;
+}
```

Comment:
This interface seems to only be used as the return type of the `useDragAndDrop` hook. I think you can drop it completely and typescript will be able to infer it.